src/components/Calendar/doubleCalendar/hooks/useHandleKeyDown.js

```diff
@@ -29,64 +29,85 @@ export default function useHandleKeyDown(
     const moveFocusedDay = useMoveFocusedDay(focusedDate, currentMonth, minDate, maxDate);
     const moveFocusedMonth = useMoveFocusedMonth(focusedDate, minDate, maxDate);
 
+    const shouldUpdateCurrentMonth = useCallback(
+        value => {
+            if (isSameMonth(value, rightCalendarMonth) || isSameMonth(value, currentMonth))
+                return false;
+            return true;
+        },
+        [currentMonth, rightCalendarMonth],
+    );
+
     const keyHandlerMap = useMemo(
         () => ({
             [UP_KEY]: () => {
                 const result = moveFocusedDay(-7);
                 setFocusedDate(result.day);
+                if (shouldUpdateCurrentMonth(result.day)) {
+                    setCurrentMonth(result.month);
+                }
```

Comment:
This logic is always repeated after setFocusedDate I think the best solution is to make a useEffect with this logic.

Reply:
yes..but it is not always the same params